storage/rocksdb/rdb_i_s.cc

```diff
@@ -771,6 +772,71 @@ static ST_FIELD_INFO rdb_i_s_global_info_fields_info[] =
   ROCKSDB_FIELD_INFO_END
 };
 
+/*
+  Support for INFORMATION_SCHEMA.ROCKSDB_COMPACTION_STATS dynamic table
+ */
+static int rdb_i_s_compact_stats_fill_table(
+    my_core::THD *thd,
+    my_core::TABLE_LIST *tables,
+    my_core::Item *cond __attribute__((__unused__)))
+{
+  DBUG_ASSERT(thd != nullptr);
+  DBUG_ASSERT(tables != nullptr);
+
+  DBUG_ENTER("rdb_i_s_global_compact_stats_table");
+
+  int ret= 0;
+
+  rocksdb::DB *rdb= rdb_get_rocksdb_db();
+  Rdb_cf_manager& cf_manager= rdb_get_cf_manager();
+  DBUG_ASSERT(rdb != nullptr);
+
+  for (auto cf_name : cf_manager.get_cf_names())
+  {
+    rocksdb::ColumnFamilyHandle* cfh;
+    bool is_automatic;
+    /*
+       Only the cf name is important. Whether it was generated automatically
+       does not matter, so is_automatic is ignored.
+       */
+    cfh= cf_manager.get_cf(cf_name.c_str(), "", nullptr, &is_automatic);
+    if (cfh == nullptr)
+      continue;
+    std::map<std::string, double> props;
+    bool bool_ret __attribute__((__unused__));
+    bool_ret = rdb->GetMapProperty(cfh, "rocksdb.cfstats", &props);
+    DBUG_ASSERT(bool_ret);
+
+    for (auto const& prop_ent : props) {
+      std::string prop_name = prop_ent.first;
+      double value = prop_ent.second;
+      std::size_t del_pos = prop_name.find('.');
+      DBUG_ASSERT(del_pos != std::string::npos);
+      std::string level_str = prop_name.substr(0, del_pos);
+      std::string type_str = prop_name.substr(del_pos + 1);
+
+      Field **field= tables->table->field;
+      DBUG_ASSERT(field != nullptr);
+      field[0]->store(cf_name.c_str(), cf_name.size(), system_charset_info);
+      field[1]->store(level_str.c_str(), level_str.size(), system_charset_info);
+      field[2]->store(type_str.c_str(), type_str.size(), system_charset_info);
+      field[3]->store(value, true);
+
+      ret |= my_core::schema_table_store_record(thd, tables->table);
```

Comment:
If ret is non-zero, should probably break out at this point since further record stores could be problematic.